src/hooks/useActiveOnboard.ts

```diff
@@ -0,0 +1,218 @@
+    const connectedChain = activeChainId     
+    const connected = [{
+      accounts: connectedAccount,
+      chains: connectedChain,
+      label: walletLabel}
+    ]
+    localStorage.setItem(
+      'connectedWallets',
+      JSON.stringify(connected)
+    )
+  }
+}
+
+// TODO: Seperate current connected wallet && ALL connected wallets
+// TODO: Handle multiple accounts connected
+/**
+ * Used for eagerly connecting on page-load when a previous wallet connection exists
+ * while also keeping track of any state updates (disconnect / switching chains)
+ * @returns tried & activated
+ */
+export function useOnboardConnect():OnboardConnectProps {
+  const [tried, setTried] = useState<boolean>(false)
+  const [activated, setActivated] = useState<boolean>(false)
+  const [{ wallet, connecting}, connect, disconnect] = useConnectWallet()
+  const [ {chains, connectedChain, settingChain}, setChain] = useSetChain()
+  const connectedWallets = useWallets()
+  const balance = connectedWallets[0]?.accounts[0]?.balance
+
+  const previouslyConnected:any = JSON.parse(
+    localStorage.getItem('connectedWallets') ?? '{}'
+  )
+  
+  const updateStorage = (newChainId:string, currentWallet:WalletState[], type:string) => {
+    const { chainId } = IsSupportedChain(newChainId)
+
+    if (type === 'init') setChain({chainId})
+    StoreOnboardState(currentWallet, chainId)
+    setActivated(true)
+  }
+
+  const chainChanged = useCallback((newChainId:string) => {
+    if (balance){
+      const isUpdated = Object.keys(balance)[0] === OnboardChainIds[parseInt(newChainId)]
+      if (isUpdated) {
+        updateStorage(newChainId, connectedWallets, 'switch')
+      }
+    }
+  }, [balance])
+
+  // TODO: Handle user rejection (patch fixed in @web3-onboard/core for now)
+
+  // eager connect
+  useEffect(() => {
+    async function connectOnboard() {
+      // Coinbase reloads instead of sending accountsChanged event, so empty storage if no active address can be found
+      if (previouslyConnected[0].label[0] === 'Coinbase'){ 
+        const isStillActive = localStorage.getItem('-walletlink:https://www.walletlink.org:Addresses')
+        if (!isStillActive){
+          localStorage.removeItem('connectedWallets')
+          setTried(true)
+          return
+        }
+      }
+      // disableModals:true for silently connecting
+      connect({ autoSelect: { label: previouslyConnected[0].label[0], disableModals: true}})
+    }
+    if (previouslyConnected[0] && !tried){
+      connectOnboard()
+      setTried(true)
+    } else if (activated || !previouslyConnected[0]) {
+      setTried(true)
+    }
+  }, [activated, tried, connect, previouslyConnected])
+
+  useEffect(() => {
+    const previousChain = previouslyConnected[0]?.chains ?? null
+    const isConnected = connectedWallets.length > 0
+    // switch chains
+    if (isConnected && previousChain && connectedChain && previousChain !== connectedChain.id) {
+      chainChanged(connectedChain.id) 
```

Comment:
here you use chainChanged and below you use updateStorage
this seems like duplicate logic, why not simply update the localstorage when connectedWallets change and its lenght > 0?

Reply:
Yes, you are right. updated in latest commit